apps/web/src/components/forms/signin.tsx

```diff
@@ -122,6 +130,15 @@ export const SignInForm = ({ className }: SignInFormProps) => {
 
         const errorMessage = ERROR_MESSAGES[result.error];
 
+        if (result.error === ErrorCode.UNVERIFIED_EMAIL) {
+          const user = await getUser({ email });
+          const token = user?.VerificationToken[user.VerificationToken.length - 1].token;
```

Comment:
Do we need a `user.VerificationToken.length > 0` check or is it implicit?

Reply:
The newly registered user will always have at least one token. Unless, it's manually deleted from the db.